apps/emails-and-messages/src/modules/shop-info/shop-info-fetcher.ts

```diff
@@ -0,0 +1,43 @@
+import { Client, gql } from "urql";
+import { ShopInfoDocument, ShopInfoFragment } from "../../../generated/graphql";
+
+gql`
+  fragment ShopInfo on Shop {
+    companyAddress {
+      country {
+        country
+        code
+      }
+      city
+      firstName
+      lastName
+      streetAddress1
+      streetAddress2
+      companyName
+      phone
+      postalCode
+      countryArea
+      cityArea
+    }
+  }
+  query ShopInfo {
+    shop {
+      ...ShopInfo
+    }
+  }
+`;
+
+export interface IShopInfoFetcher {
+  fetchShopInfo(): Promise<ShopInfoFragment | null>;
+}
+
+export class ShopInfoFetcher implements IShopInfoFetcher {
```

Comment:
where do you use it? Isnt it copied from invoices?

Reply:
Used in the code building the initial configuration